aws-api-appsync/src/main/java/com/amplifyframework/api/aws/SelectionSet.java

```diff
@@ -276,16 +276,20 @@ private SelectionSet wrapPagination(SelectionSet node) {
          */
         private static boolean isCustomType(@NonNull Field field) {
             Class<?> cls = getClassForField(field);
-            if (Model.class.isAssignableFrom(cls) || Enum.class.isAssignableFrom(cls)) {
+            if (cls.getClassLoader() == null || isDateType(cls)) {
```

Comment:
Could you add a comment explaining how this works?  (essentially what you've put in PR description)

Reply:
@richardmcclellan  sure will add!